app/controllers/course/material/folders_controller.rb

```diff
@@ -0,0 +1,52 @@
+class Course::Material::FoldersController < Course::ComponentController
+  load_and_authorize_resource :material_folder, through: :course, parent: false,
```

Comment:
that's a call you need to make. either present a generic folders association, and filter for materials, or have two separate associations. tell me which you think is better.